lib/CANNON-shape2mesh.js

```diff
@@ -89,39 +100,39 @@ CANNON.shape2mesh = function(body){
                         v0.vadd(shape.pillarOffset, v0);
                         v1.vadd(shape.pillarOffset, v1);
                         v2.vadd(shape.pillarOffset, v2);
-                        geometry.vertices.push(
+                        vertices.push(
                             new THREE.Vector3(v0.x, v0.y, v0.z),
                             new THREE.Vector3(v1.x, v1.y, v1.z),
                             new THREE.Vector3(v2.x, v2.y, v2.z)
                         );
-                        var i = geometry.vertices.length - 3;
-                        geometry.faces.push(new THREE.Face3(i, i+1, i+2));
+                        var i = vertices.length - 3;
+                        faces.push(i, i+1, i+2);
```

Comment:
Looking at `geometry.setFromPoints` https://github.com/mrdoob/three.js/blob/3119830e3632df95901af6e9ec2d0ac19bbc4789/src/core/BufferGeometry.js#L273-L288
creating all those intermediate THREE.Vector3 for vertices are actually useless.
We probably want to create right away the position array
```
var position = [];
for
  for
    for
      ...
      position.push(v0.x, v0.y, v0.z, v1.x, v1.y, v1.z, v2.x, v2.y, v2.z);
```
and in `createBufferGeometry` instead of using 
```
geometry.setFromPoints(vertices);
````
do:

```
    function createBufferGeometry(position, faces) {
      var geometry = new THREE.BufferGeometry();
      geometry.setAttribute( 'position', new Float32BufferAttribute( position, 3 ) );
```

and of course replace
```
                        var i = vertices.length - 3;
```
by
```
                        var i = position.length / 3 - 3;
```

What do you think?

Reply:
Thanks @vincentfretin you are absolutely right.

Initially I was trying to keep the code change from a previous version to a minimum, to avoid risk of introducing bugs with the switch across to the new interface.

But now I've got it working, the code absolutely can be refactored as you suggest.  This will make it simpler & should perform better as well.